cli/pkg/kctrl/cmd/package/release/helm_openapi_schema_gen.go

```diff
@@ -0,0 +1,361 @@
+				}
+				for _, item := range calculatedProperties.Items {
+					if item.Key == itemsKey {
+						properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+					}
+				}
+			case yaml3.ScalarNode:
+				val, err := h.getDefaultValue(v.Tag, v.Value)
+				if err != nil {
+					return nil, err
+				}
+				defaultVals = append(defaultVals, val)
+			case yaml3.AliasNode:
+				calculatedProperties, err := h.calculateProperties(nil, v.Alias)
+				if err != nil {
+					return nil, err
+				}
+				for _, item := range calculatedProperties.Items {
+					if item.Key == itemsKey {
+						properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+					}
+				}
+			default:
+				return nil, fmt.Errorf("Unrecognized type %T", v.Kind)
+			}
+		}
+
+		if len(value.Content) != 0 {
+			var itemsProperties *Map
+			switch value.Content[0].Kind {
+			case yaml3.MappingNode, yaml3.AliasNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "object"},
+					&MapItem{Key: propertiesKey, Value: properties}}}
+			case yaml3.SequenceNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "array"},
+					&MapItem{Key: defaultKey, Value: "[]"},
+					&MapItem{Key: itemsKey, Value: properties}}}
+			case yaml3.ScalarNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: h.openAPIType(value.Content[0].Tag, value.Content[0].Value)}}}
+			}
+			apiKeys = append(apiKeys, &MapItem{Key: itemsKey, Value: itemsProperties})
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVals})
+		sort.Sort(byKey(apiKeys))
+		if key == nil {
+			return &Map{Items: apiKeys}, nil
+		}
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.ScalarNode:
+		var apiKeys []*MapItem
+		description, isPresent := h.getDescriptionFromNode(key)
+		if isPresent {
+			apiKeys = append(apiKeys, &MapItem{Key: descriptionKey, Value: description})
+		}
+
+		defaultVal, err := h.getDefaultValue(value.Tag, value.Value)
+		if err != nil {
+			return nil, err
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: typeKey, Value: h.openAPIType(value.Tag, value.Value)})
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVal})
+		if value.Tag == floatTag {
+			apiKeys = append(apiKeys, &MapItem{Key: formatKey, Value: floatVal})
+		}
+
+		sort.Sort(byKey(apiKeys))
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.AliasNode:
+		return h.calculateProperties(key, value.Alias)
+	default:
+		return nil, fmt.Errorf("Unrecognized type %T", value.Kind)
+	}
+	return nil, nil
+}
+
+func (h HelmValuesSchemaGen) getDescriptionFromNode(node *yaml3.Node) (string, bool) {
+	if node == nil || node.HeadComment == "" {
+		return "", false
+	}
+
+	comment := node.HeadComment
+	comment = strings.ReplaceAll(comment, "\n##", "")
+	comment = strings.ReplaceAll(comment, "\n#", "")
```

Comment:
I am a bit unclear about what we are doing over here 🤔

Reply:
Sometimes in helm values.yaml, comment is divided into multi line, we are converting them to a single line